Mathlib/RingTheory/PrincipalIdealDomain.lean

```diff
@@ -480,30 +480,39 @@ open Set Ideal
 variable (R) [CommRing R]
 
 /-- `nonPrincipals R` is the set of all ideals of `R` that are not principal ideals. -/
+@[deprecated "Write { I : Ideal R | ¬I.IsPrincipal } explicitely instead." (since := "2025-08-16")]
```

Comment:
You can make this an abbrev or notation, or something similar, but it seems a good idea to keep a short name.

Reply:
So I think the best is having
```lean
/-- `nonPrincipals R` is the set of all ideals of `R` that are not principal ideals. -/
abbrev nonPrincipals := {I : Ideal R | ¬I.IsPrincipal}
```
And you can remove `nonPrincipals_def`. Moreover everything should be in the `Ideal` namespace.